homeassistant/components/motionmount/number.py

```diff
@@ -0,0 +1,78 @@
+"""Support for MotionMount numeric control."""
+import motionmount
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import MotionMountEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Vogel's MotionMount from a config entry."""
+    mm: motionmount.MotionMount = hass.data[DOMAIN][entry.entry_id]
+
+    unique_id = format_mac(mm.mac.hex())
```

Comment:
We need to handle the EMPTY_MAC case here.
I'd move the `base_unique_id` generation into the entity.py base class so you can do something like

```python
base_unique_id = format_mac(mm.mac.hex())
if base_unique_id == EMPTY_MAC:
   self._base_unique_id = config_entry.entry_id
else:
  self._base_unique_id = base_unique_id
```

Than you can do something like
```python
self._attr_unique_id = f"{self._base_unique_id}-extension"
```

Reply:
This is a smart move indeed. It will also help with the additional entities that will be added later on.